include/onnxruntime/core/session/onnxruntime_cxx_api.h

```diff
@@ -41,9 +37,23 @@ struct Exception : std::exception {
   OrtErrorCode code_;
 };
 
-// This Macro is to make it easy to generate overloaded methods for all of the various OrtRelease* functions for every Ort* type
+// This is used internally by the C++ API. This class holds the global variable that points to the OrtApi, it's in a template so that we can define a global variable in a header and make
+// it transparent to the users of the API.
+template <typename T>
```

Comment:
The definition of T is unclear as the usage is unclear. It looks like T can be anything?

Reply:
Yep, it can be anything. It just needs to be a template since we can't do inline variables in pre C++17. That's why I say it's used internally, aka.. ignore it if you're using the header.